core/src/main/scala/org/apache/spark/storage/BlockManagerId.scala

```diff
@@ -132,10 +133,15 @@ private[spark] object BlockManagerId {
     getCachedBlockManagerId(obj)
   }
 
-  val blockManagerIdCache = new ConcurrentHashMap[BlockManagerId, BlockManagerId]()
+  val blockManagerIdCache = CacheBuilder.newBuilder()
```

Comment:
is it thread-safe?

Reply:
I think it is thread-safe which i refer from:
https://google.github.io/guava/releases/22.0/api/docs/com/google/common/cache/LoadingCache.html
and https://stackoverflow.com/questions/11124856/using-guava-for-high-performance-thread-safe-caching